### DISCUSSION TITLE
Can one use a different LaTeX engine with MyST?

### DISCUSSION THREAD

**Discussion by markusritschel** [executablebooks/meta#1154] [Q&A]
Is it possible to use another LaTeX engine such as [Tectonic](https://tectonic-typesetting.github.io/en-US/) with MyST?

**Comment by rowanc1** [comment]
It is certainly possible to develop, but it isn't hooked up at the moment! From the landing page of Tectonic, this might actually be something that we should switch to wholesale. Right now we are using `XeTeX` and are not doing dependency management. I would certainly welcome any pointers or help if you are familiar with these tools!

**Reply by markusritschel** [reply]
Hey @rowanc1,
Thanks for the immediate reply.
Great to hear you consider it as an option to implement. I have been using Tectonic now for a few years and I like it a lot.
I did a simple test by `latexmk` with `tectonic`, and it works :-)
If you want, I can open a pull request with these changes.?

**Reply by rowanc1** [reply]
Very open to a PR! The `latexmk` line is here:
https://github.com/executablebooks/mystjs/blob/main/packages/jtex/src/tex/export.ts#L12
Not overly sophisticated. :)

I think it would be good to use tectonic if it is available, similar to this:
https://github.com/executablebooks/mystjs/blob/main/packages/myst-cli/src/build/pdf/utils.ts#L5

And then fall back to the existing command.

Any help updating docs in light of this change would also be very much appreciated.

**Reply by markusritschel** [reply]
Ok, great. Will try. Not an expert in Javascript or Typescript ;-)

**Reply by markusritschel** [reply]
In `packages/jtex/src/tex/export.ts`, I would like to include an option with an _if_ statement, making `tectonic` the default with a fallback to `latexmk`. 
However, I cannot `import {isLatexmkAvailable, isTectonicAvailable } from '../../../myst-cli/src/build/pdf/utils.ts';`. I would need to add 
```json
{
  "compilerOptions": {
    "allowJs": true,
    "allowImportingTsExtensions": true
  }
}
 ```
 to the _tsconfig.json_.
 However, doing so would make the build process fail.
 
 I currently circumnavigate this by duplicating the functions `isLatexmkAvailable` and `isTectonicAvailable` from _/packages/myst-cli/src/build/pdf/utils.ts_ in _/packages/jtex/src/tex/export.ts_. This doesn't seem elegant, though.
 Any ideas?

**Reply by rowanc1** [reply]
In these packages imports are designed to go from the component packages (e.g. jtex in this case) to be imported into the `myst-cli`. It isn't possible to import across the package boundaries (i.e. `../../../myst-cli/src`).

There are a couple ways to go - moving the  `isLatexmkAvailable` and `isTectonicAvailable` down to jtex, or moving those compile functions up. I think I would prefer to move them up and remove them from jtex, leaving the scope of that repo to only be about templating, not compiling and choosing these options.

I will take a look at your pr and continue the conversation over there!

**Comment by markusritschel** [comment]
Opened [PR](https://github.com/executablebooks/mystmd/pull/1093).
